File: src/main/java/org/itmo/isLab1/dragonheads/DragonHeadService.java
package org.itmo.isLab1.dragonheads;

import org.springframework.stereotype.Service;
import org.itmo.isLab1.common.framework.CrudService;
import org.itmo.isLab1.dragonheads.dto.*;
import org.itmo.isLab1.dragonheads.mapper.DragonHeadMapper;
import org.itmo.isLab1.dragonheads.policy.DragonHeadPolicy;
import org.itmo.isLab1.events.EventService;
import org.itmo.isLab1.users.UserService;

@Service
public class DragonHeadService
    extends CrudService<
        DragonHead,
        DragonHeadRepository,
        DragonHeadMapper,
        DragonHeadPolicy,
        DragonHeadDto,
        DragonHeadCreateDto,
        DragonHeadUpdateDto> {

    public DragonHeadService(
        DragonHeadRepository repository,
        DragonHeadMapper mapper,
        DragonHeadPolicy policy,
        UserService userService,
        EventService<DragonHead> eventService
    ) {
        super(repository, mapper, policy, userService, eventService);
    }
}

File: frontend/src/services/DragonHeadService.ts
import { DragonHeadCreateDto } from '@/interfaces/dto/dragonheads/DragonHeadCreateDto'
import { DragonHeadDto } from '@/interfaces/dto/dragonheads/DragonHeadDto'
import { DragonHeadUpdateDto } from '@/interfaces/dto/dragonheads/DragonHeadUpdateDto'
import DragonHeadController from '@/controllers/DragonHeadController'
import CrudService from '@/interfaces/crud/CrudService';

class DragonHeadService extends CrudService<DragonHeadDto, DragonHeadCreateDto, DragonHeadUpdateDto> {
    constructor() {
        super("DragonHead", DragonHeadController);
    }

    handleObjectKill(object: DragonHeadDto) {
    }
}

export default new DragonHeadService()